labtech/tasks.py

```diff
@@ -255,4 +251,9 @@ def find_tasks_in_param(param_value: Any, searched_coll_ids: Optional[Set[int]]
             for item in param_value.values()
             for task in find_tasks_in_param(item, searched_coll_ids)
         ]
-    return []
+    elif isinstance(param_value, cast(UnionType, ParamScalar)):
+        return []
+
+    # This should be impossible.
+    msg = f"Unexpected type {type(param_value).__qualname__} encountered in task parameter value."
+    raise AssertionError(msg)
```

Comment:
I believe `AssertionError` is only intended for use by `assert` statements, so I suggest we make this a `TaskError` for consistency with other errors in this file.

```suggestion
    raise TaskError(msg)
```